### PR TITLE
Event and Organization Action item layouts need to be the same

### LINKED ISSUE #1930: Event and Organization Action item layouts need to be the same
**Describe the bug**

Event and Organization Action item layouts need to be the same

**To Reproduce**

1. Create action items for an event. Observe the screens
2. Create action items for an organization Observe the screens
3. The screens and modals are different

**Design**

1. Before starting the work, create a figma design that can be reviewed.
2. The styling must be in keeping with our existing esthetic

**Expected behavior**

1. The layouts and modals must match
1. We must have valid tests to ensure that we don't have this future divergence. All tests must pass.

**Actual behavior**

- See above

**Screenshots**

1. Organization Action Items Screen

    image

1. Event Action Items Screen

    image

1. Event action item manage modal doesn't show the assignee

    image

1. Action item headings should be "Notes" and "Completion Notes"

    image

**Additional details**

- N/A

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 

- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Chaitanya1672**
**What kind of change does this PR introduce?**
bugfix

**Issue Number:**

Fixes #1930 

**Did you add tests for your changes?**
Yes

**Video:**

https://github.com/PalisadoesFoundation/talawa-admin/assets/100834122/189941e4-dea5-4014-b3eb-daf925560255

**Snapshots:**
image
image
image
image
image
image

**Summary**
1) Changed the headers from `Pre Completion Notes & Post Completion Notes` to `Notes` and `Completion Notes`.
2) Updated the checkbox size on the Action Items page to match the size of other options.
3) Fixed the bug in the update action modal on the organization action items page where data such as `Assignee` was not prepopulating.

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

## Summary by CodeRabbit

- **Updates**
  - Renamed labels related to completion notes for better clarity across all supported languages (English, French, Hindi, Spanish, and Chinese).
  - Added new labels for assignment details and action item status.

- **Localization**
  - Updated translation files to include new keys and revised existing ones for better consistency and accuracy.

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@Chaitanya1672 Please resolve the failed linting check.

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. The layouts and modals must match as requested in the original issue. 
    1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1930
1. Use the Organization layout as a guide. 
    1. Some elements won't be needed in the event action items such as search and filtering at this time.
    2. The "create action items" button should be renamed "create" and positioned in an aesthetically pleasing location
        image

**Comment by Chaitanya1672** [comment]
Hi @palisadoes 
Please verify the changes below and please do let me know the other additional modifications needed (e.g. modal/options, etc.).
image

checkbox change below
image

**Comment by palisadoes** [comment]
They are still not the same.

1. One has links to the assignee profile - this is preferred.
2. One has a number value in the first column - this is preferred
3. The options column is different. One uses icons - this is preferred

File: src/components/ActionItems/ActionItemsContainer.test.tsx [hunk 1]
```diff
@@ -86,6 +86,35 @@ describe('Testing Action Item Categories Component', () => {
...
+    expect(screen.getByText(translations.assignee)).toBeInTheDocument();
+    expect(
+      screen.getByText(translations.actionItemCategory),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByText(translations.preCompletionNotes),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByText(translations.postCompletionNotes),
+    ).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (Chaitanya1672) [reply src/components/ActionItems/ActionItemsContainer.test.tsx:100]: Translations are changed, but the key name is kept the same. For example: `preCompletionNotes: Notes`

File: src/components/ActionItems/ActionItemsContainer.test.tsx [hunk 2]
```diff
@@ -667,4 +696,34 @@ describe('Testing Action Item Categories Component', () => {
...
```
[the remark this answers is not in the document]
Comment (Chaitanya1672) [reply src/components/ActionItems/ActionItemsContainer.test.tsx:726]: Same here 
Translations are changed, but the key name is kept the same. For example: `preCompletionNotes: Notes`

File: src/components/EventManagement/EventActionItems/EventActionItems.module.css [hunk 3]
```diff
@@ -128,18 +136,33 @@
...
+  width: 100%;
+  font-size: 16px;
+  color: var(--bs-white);
+  outline: none;
+  font-weight: 600;
+  cursor: pointer;
+  transition:
+    transform 0.2s,
+    box-shadow 0.2s;
+  width: 100%;
```
[the remark this answers is not in the document]
Comment (Chaitanya1672) [reply src/components/EventManagement/EventActionItems/EventActionItems.module.css:161]: By default, Bootstrap provides a hover effect to button.

File: src/screens/OrganizationActionItems/ActionItemUpdateModal.tsx [hunk 4]
```diff
@@ -107,36 +109,33 @@ const ActionItemUpdateModal: React.FC<InterfaceActionItemCreateModalProps> = ({
...
+                  /* istanbul ignore next */ (date: Dayjs | null): void => {
+                    /* istanbul ignore next */
+                    if (date) {
+                      setCompletionDate(date?.toDate());
                     }
                   }
-                />
-              </div>
+                }
+              />
```
[the remark this answers is not in the document]
Comment (Chaitanya1672) [reply src/screens/OrganizationActionItems/ActionItemUpdateModal.tsx:138]: I'm not sure of the reason for using this, as it has been there from the start. I will keep it as it is, but let me know if any changes are needed.

File: src/components/ActionItems/ActionItemsContainer.tsx [hunk 5]
```diff
@@ -237,29 +253,44 @@ function actionItemsContainer({
...
-                  className="align-self-center text-body-secondary"
+                  className="d-flex align-items-center justify-content-center text-body-secondary"
                 >
-                  {`${actionItem.assignee.firstName} ${actionItem.assignee.lastName}`}
+                  <Link
+                    to={`/member/${actionItem?.event?._id}`}
+                    state={{ id: index }}
+                    className={styles.membername}
+                  >
+                    {`${actionItem.assignee.firstName} ${actionItem.assignee.lastName}`}
```
[the remark this answers is not in the document]
Comment (Chaitanya1672) [reply src/components/ActionItems/ActionItemsContainer.tsx:277]: We are passing the key as `key={actionItem._id}`, and the index is just used for serial numbers.

File: src/screens/OrganizationActionItems/ActionItemUpdateModal.tsx [hunk 6]
```diff
@@ -75,16 +75,18 @@ const ActionItemUpdateModal: React.FC<InterfaceActionItemCreateModalProps> = ({
...
+                    const currMemberName = `${member.firstName} ${member.lastName}`;
+                    if (currMemberName !== formState.assignee) {
+                      return (
+                        <option key={index} value={member._id}>
+                          {`${member.firstName} ${member.lastName}`}
+                        </option>
+                      );
+                    }
+                  },
+                )}
```
[the remark this answers is not in the document]
Comment (Chaitanya1672) [reply src/screens/OrganizationActionItems/ActionItemUpdateModal.tsx:89]: fixed in latest commit

**Comment by palisadoes** [comment]
1. The coverage tests are failing.
2. Please get the test code coverage of the patch higher so that we can review and merge.

**Comment by palisadoes** [comment]
Please attach a video showing the same layouts

**Comment by Chaitanya1672** [comment]
Hi @palisadoes, I have added the video, can you please review the code.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**